modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,49 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
 
-				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
+				var images = document.querySelectorAll( 'img.wp-image-' + media[ i ].id ),
+					media_sources = media[ i ].media_details.sources;
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
-					var srcset = images[j].getAttribute( 'srcset' );
+
+					if ( ! media_sources || ! media_sources['image/jpeg'] ) {
+						continue;
+					}
```

Comment:
Great idea! If there is no JPEG for the `full` image, it means no image for that attachment will have a JPEG, as JPEGs are always created unless it is disabled for the entire attachment (e.g. if a site is configured to only generate WebP).

Would be good to add a comment though to explain this, otherwise it may not make sense to someone without the context.
```suggestion
					// If the full image has no JPEG version available, no sub-size will have JPEG available either.
					if ( ! media_sources || ! media_sources['image/jpeg'] ) {
						continue;
					}
```

Reply:
Thank you for the review. This feedback is addressed.